include/bluetooth/services/fast_pair.h

```diff
@@ -30,18 +37,82 @@ enum bt_fast_pair_adv_mode {
 
 	/** Fast Pair not discoverable advertising, hide UI indication. */
 	BT_FAST_PAIR_ADV_MODE_NOT_DISCOVERABLE_HIDE_UI_IND,
+
 	/** Number of Fast Pair advertising modes. */
 	BT_FAST_PAIR_ADV_MODE_COUNT
 };
 
+/** @brief Fast Pair advertising battery mode. Used to generate advertising packet.
+ *
+ * Battery data can be included in advertising packet only if the Fast Pair Provider is in Fast Pair
+ * not discoverable advertising mode. To prevent tracking, the Fast Pair Provider should not include
+ * battery data in the advertising packet all the time.
+ */
+enum bt_fast_pair_adv_battery_mode {
+	/** Do not advertise battery data. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_NONE,
+
+	/** Show battery data UI indication. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_SHOW_UI_IND,
+
+	/** Hide battery data UI indication. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_HIDE_UI_IND,
+
+	/** Number of Fast Pair advertising battery modes. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_COUNT
+};
+
+/** @brief Index of battery value. */
+enum bt_fast_pair_battery_value_idx {
+	/** Left bud. */
+	BT_FAST_PAIR_BATTERY_VALUE_IDX_LEFT_BUD = 0,
+
+	/** Right bud. */
+	BT_FAST_PAIR_BATTERY_VALUE_IDX_RIGHT_BUD = 1,
+
+	/** Case. */
+	BT_FAST_PAIR_BATTERY_VALUE_IDX_BUD_CASE = 2,
+
+	/** Number of battery values. */
+	BT_FAST_PAIR_BATTERY_VALUE_IDX_COUNT
+};
+
+/** @brief Fast Pair advertising config. Used to generate advertising packet. */
+struct bt_fast_pair_adv_config {
+	/** Fast Pair advertising mode. */
+	enum bt_fast_pair_adv_mode adv_mode;
+
+	/** Fast Pair advertising battery mode. */
+	enum bt_fast_pair_adv_battery_mode adv_battery_mode;
+};
+
+/** @brief Fast Pair battery value related to a single battery */
+struct bt_fast_pair_battery_value {
```

Comment:
`bt_fast_pair_battery_status` (it's value + charging_status)

Reply:
![image](https://user-images.githubusercontent.com/86837740/200822140-5c47c16f-8e6c-4dc9-92fe-a47a6d703f08.png)
I wanted to be consistent with the spec. We now have a `bt_fast_pair_battery_value` that consists of status and level. I like it more than proposed `bt_fast_pair_battery_status`.

Reply:
To build on top of suggestions from the https://github.com/nrfconnect/sdk-nrf/pull/9058/files#r1011542743 discussion, we can call it ```bt_fast_pair_battery_comp_desc``` to indicate that the structure is a descriptor of a battery component